### PR TITLE
[Merged by Bors] - feat(Topology/LocallyClosed): Define locally closed sets

### PR DESCRIPTION
**Pull request by erdOne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/LocallyClosed.lean [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+  exact hf.closure_preimage_subset s
+
+lemma coborder_preimage (hf : IsOpenMap f) (hf' : Continuous f) (s : Set β) :
+    coborder (f ⁻¹' s) = f ⁻¹' (coborder s) :=
+(hf.coborder_preimage_subset s).antisymm (hf'.preimage_coborder_subset s)
+
+protected
+lemma OpenEmbedding.coborder_preimage (hf : OpenEmbedding f) (s : Set β) :
+    coborder (f ⁻¹' s) = f ⁻¹' (coborder s) :=
+coborder_preimage hf.isOpenMap hf.continuous s
```
Comment (dagurtomas) [inline Mathlib/Topology/LocallyClosed.lean:91]:
```suggestion
  (hf.coborder_preimage_subset s).antisymm (hf'.preimage_coborder_subset s)

protected
lemma OpenEmbedding.coborder_preimage (hf : OpenEmbedding f) (s : Set β) :
    coborder (f ⁻¹' s) = f ⁻¹' (coborder s) :=
  coborder_preimage hf.isOpenMap hf.continuous s
```

File: Mathlib/Topology/LocallyClosed.lean [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+Authors: Andrew Yang
+-/
+import Mathlib.Topology.LocalAtTarget
+
+/-!
+# Locally closed sets
+
+## Main results
+- `IsLocallyClosed`: Predicate saying that a set is locally closed
+- `isLocallyClosed_tfae`:
```
Comment (fpvandoorn) [inline Mathlib/Topology/LocallyClosed.lean:13]:
```suggestion
## Main definitions

* `IsLocallyClosed`: Predicate saying that a set is locally closed

## Main results

* `isLocallyClosed_tfae`:
```

File: Mathlib/Topology/LocallyClosed.lean [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  (hf.coborder_preimage_subset s).antisymm (hf'.preimage_coborder_subset s)
+
+protected
+lemma OpenEmbedding.coborder_preimage (hf : OpenEmbedding f) (s : Set β) :
+    coborder (f ⁻¹' s) = f ⁻¹' (coborder s) :=
+  coborder_preimage hf.isOpenMap hf.continuous s
+
+lemma isClosed_preimage_val : IsClosed (s ↓∩ t) ↔ s ∩ closure (s ∩ t) ⊆ t := by
+  rw [← closure_eq_iff_isClosed, embedding_subtype_val.closure_eq_preimage_closure_image,
+    ← (Set.image_injective.mpr Subtype.val_injective).eq_iff, Subtype.image_preimage_coe,
```
Comment (fpvandoorn) [inline Mathlib/Topology/LocallyClosed.lean:95]:
```suggestion
    ← Subtype.val_injective.image_injective.eq_iff, Subtype.image_preimage_coe,
```
Please also move this lemma to `Constructions`

File: Mathlib/Topology/LocallyClosed.lean [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    ← Set.union_diff_right, Set.union_comm, ← closure_eq_self_union_frontier]
+
+lemma coborder_eq_univ_iff :
+    coborder s = Set.univ ↔ IsClosed s := by
+  simp [coborder, Set.diff_eq_empty, closure_subset_iff_isClosed]
+
+alias ⟨_, IsClosed.coborder_eq⟩ := coborder_eq_univ_iff
+
+lemma coborder_eq_compl_frontier_iff :
+    coborder s = (frontier s)ᶜ ↔ IsOpen s := by
```
Comment (fpvandoorn) [inline Mathlib/Topology/LocallyClosed.lean:65]: Can you add a lemma stating `Disjoint (frontier s) s ↔ IsOpen s`? Then this proof can become
```lean
  simp_rw [coborder_eq_union_frontier_compl, union_eq_right, subset_compl_iff_disjoint_left, 
    disjoint_frontier_iff_isOpen]
```

**Review by fpvandoorn — COMMENTED** [review]
Looks mostly good to me.

Please open the `Set` namespace and remove `Set.` prefixes.

File: Mathlib/Topology/LocallyClosed.lean [hunk 5]
```diff
@@ -8,11 +8,15 @@ import Mathlib.Topology.LocalAtTarget
...
+
+* `IsLocallyClosed`: Predicate saying that a set is locally closed
+
 ## Main results
-- `IsLocallyClosed`: Predicate saying that a set is locally closed
-- `isLocallyClosed_tfae`:
+
+* `isLocallyClosed_tfae`:
   A set `s` is locally closed if one of the equivalent conditions below hold
-  1. It is the intersection of some open set and some closed set.
```
Comment (fpvandoorn) [inline Mathlib/Topology/LocallyClosed.lean]: You removed `set.` three times in a comment

**Comment by fpvandoorn** [comment]
One more comments, otherwise LGTM

bors d+

**Comment by erdOne** [comment]
bors merge

**Pull request closed without merging**